packages/runtime-core/__tests__/apiWatch.spec.ts

```diff
@@ -1564,6 +1564,20 @@ describe('api: watch', () => {
     expect(cb).toHaveBeenCalledTimes(1)
   })
 
+  // #3341
+  test('watchEffect should allow multiple onCleanup calls', async () => {
```

Comment:
This test name and changes of this PR don't seem to match. In fact, `onCleanup` already supports multiple calls, and this PR is about deduplicating the cleanup function.